lib/tdf3/src/tdf.ts

```diff
@@ -980,58 +965,122 @@ async function unwrapKey({
       };
     }
 
-    // Create a PoP token by signing the body so KAS knows we actually have a private key
-    // Expires in 60 seconds
     const httpReq = await authProvider.withCreds(buildRequest('POST', url, requestBody));
+    const {
+      data: { entityWrappedKey, metadata },
+    } = await axios.post(httpReq.url, httpReq.body, { headers: httpReq.headers });
+
+    const key = Binary.fromString(base64.decode(entityWrappedKey));
+    const decryptedKeyBinary = await cryptoService.decryptWithPrivateKey(
+      key,
+      ephemeralEncryptionKeys.privateKey
+    );
+
+    return {
+      key: new Uint8Array(decryptedKeyBinary.asByteArray()),
+      metadata,
+    };
+  }
+
+  // Get unique split IDs to determine if we have an OR or AND condition
+  const splitIds = new Set(Object.keys(splitPotentials));
+
+  // If we have only one split ID, it's an OR condition
+  if (splitIds.size === 1) {
```

Comment:
This should be the same as the case where there are multiple splitIds, this branch should not be required (maybe as an optimization to avoid having a singleton all promise but I doubt that makes a big difference).

Do you have a link to the failing test this was written to address?

Reply:
split id tests are in xtest.  test_autoconfigure_double_kas, test_autoconfigure_two_kas

some more have been added and are pending.
no unit tests for this; couldn't find anything to build upon